### ISSUE TITLE
How to set source batch and target batch to fetch data as the data order in ImageSets.

### ISSUE THREAD

**Issue by RhythmOfTheRain-Byte** [open-mmlab/OpenPCDet#1590]
Excues me，could you help me solve the problem.
I want to set both source batch and target batch to fetch data as the order in ImageSets. But When shuffle is set to false, this goal still cannot be achieved. May I ask how to solve it.

dataloader=DataLoader in pcdet/datasets/__ init__. py（
Dataset, batch size=batch size, pin_memory=True, num_workers=workers,
Shuffle=(sampler is None) and training, collate_fn=dataset. collate_batch,
#Droplast=True, sampler=sampler, timeout=0
Droplast=False, sampler=sampler, timeout=0
）
For example , the ImageSets is the same for both source dataset and target dataset and part of ImageSets is as fellows :
2019-09-11_18-41-27,02580
2019-09-11_18-56-46,01920
2018-02-17_10-31-08,00400
2019-09-11_19-09-25,00420
2019-09-11_17-36-19,00480
2018-02-05_12-15-36,00000
2019-09-11_17-45-16,01560
2019-09-11_19-13-44,00180
2019-09-11_17-20-55,00420
2018-02-05_12-06-04,00300
2019-09-11_17-45-16,00000
2019-09-11_18-38-35,00150
The batch size is set as 4, when the train is executated:
the frame id in source_batch is :
0:
'2018-02-04_21-42-09_00100'
1:
'2018-02-09_11-20-33_00100'
2:
'2019-09-11_18-56-46_00060'
3:
'2018-12-13_15-31-33_01200'
while the frame id in target_batch is :
0:
'2018-12-13_15-39-55_00300'
1:
'2018-12-22_15-01-45_02600'
2:
'2019-05-02_21-18-08_03390'
3:
'2019-05-02_18-30-47_01230'
how to set the same frame id in source_batch and target_batch for each batch.
The core code is as fellows:

source_set, source_loader, source_sampler = build_dataloader(
        dataset_cfg=cfg.DATA_CONFIG,
        class_names=cfg.CLASS_NAMES,
        batch_size=args.batch_size,
        dist=dist_train, workers=args.workers,
        logger=logger,
        training=True,
        merge_all_iters_to_one_epoch=args.merge_all_iters_to_one_epoch,
        total_epochs=args.epochs,
        seed=666 if args.fix_random_seed else None #为了使得batch能按照顺序取样新增行
    )
def build_dataloader(dataset_cfg, class_names, batch_size, dist, root_path=None, workers=4,seed=None,
                     logger=None, training=True, merge_all_iters_to_one_epoch=False, total_epochs=0):

    dataset = __all__[dataset_cfg.DATASET](
        dataset_cfg=dataset_cfg,
        class_names=class_names,
        root_path=root_path,
        training=training,
        logger=logger,
    )

    if merge_all_iters_to_one_epoch:
        assert hasattr(dataset, 'merge_all_iters_to_one_epoch')
        dataset.merge_all_iters_to_one_epoch(merge=True, epochs=total_epochs)

    if dist:
        if training:
            sampler = torch.utils.data.distributed.DistributedSampler(dataset)
        else:
            rank, world_size = common_utils.get_dist_info()
            sampler = DistributedSampler(dataset, world_size, rank, shuffle=False)
    else:
        sampler = None
    dataloader = DataLoader(
        dataset, batch_size=batch_size, pin_memory=True, num_workers=workers,
        # shuffle=(sampler is None) and training, collate_fn=dataset.collate_batch,
        shuffle= False, collate_fn=dataset.collate_batch,
        drop_last=False, sampler=sampler, timeout=0,
        # drop_last=True, sampler=sampler, timeout=0
        worker_init_fn=partial(common_utils.worker_init_fn, seed=seed)
    )

    return dataset, dataloader, sampler
